stdlib/3/builtins.pyi

```diff
@@ -738,10 +738,10 @@ class list(MutableSequence[_T], Generic[_T]):
         def __class_getitem__(cls, item: Any) -> GenericAlias: ...
 
 class dict(MutableMapping[_KT, _VT], Generic[_KT, _VT]):
-    # NOTE: Keyword arguments are special. If they are used, _KT must include
-    #       str, but we have no way of enforcing it here.
     @overload
-    def __init__(self, **kwargs: _VT) -> None: ...
+    def __init__(self: Dict[Any, Any]) -> None: ...
+    @overload
+    def __init__(self: Dict[str, _VT], **kwargs: _VT) -> None: ...
     @overload
     def __init__(self, map: SupportsKeysAndGetItem[_KT, _VT], **kwargs: _VT) -> None: ...
```

Comment:
This overload and the next should also require `_KT` to include str if kwargs is nonempty. That's probably not worth bothering with though.